Design/purity-checker-proposal.md

```diff
@@ -0,0 +1,103 @@
+## !First Concept! for the Purity Checker
+
+We want a call of `checkValidity()` on an arbitrary `ExpEmbedding` node to return `true` if all `assert` statements present contain only pure expressions, and `false` as soon as one of them is impure. We split the logic into traversal and validation, currently only focussing on the validation of expressions appearing inside of `assert` nodes.
```

Comment:
This is very specific.  What we want is to introduce a notion of validity that is established after the translation to ExpEmbedding, and assert statements with impure conditions are an example of a violation.